### DISCUSSION TITLE
Start defaultValues ​​after change state

### DISCUSSION THREAD

**Discussion by dajimene** [react-hook-form/react-hook-form#3222] [Q&A]
Dear s, I need your help, I am using the react-hook-form library for the following scenario:
1.- I have to start two variables, one of them is modified according to the button I press to change its color.
const defaultValues ​​= {customerid: '1', type: 'HOME',};
const {control, handleSubmit, errors, setValue, register} = useForm ({defaultValues});
2.- When I press the buttons 1 OR 2 the state (TYPE) is changed, I verify it when I execute handleSubmit
My problem occurs when pressing buttons 1 OR 2 the defaultValues ​​variable is not updated and that depends on which buttons are painted.
Your help on how to solve my problem I leave the coding

https://snack.expo.io/cyK9JB1uo

**Comment by bluebill1049** [comment]
- use Controller for those inputs
- use reset API to reset the defaultValues for your entire form

**Reply by dajimene** [reply]
Friend you tell me to do a reset what I can see the documentation is done to controlled fields I am starting a variable in defaultValues ​​that is not associated with any input, as a change in the state button onPress = {() => reset ({ type: "DEPARTMENT"}) if type has no access outside of defaultValues

Did you check the code ???, it is feasible what I am requesting with your library

**Reply by dajimene** [reply]
if I create a useState for the state change every time I press the button there I get the expected result but I understand that it should not be affecting manually if react-hook-form should do it automatically

const [info, setInfo] = React.useState({});

const setAddressTypee = (ty) => {
setInfo(ty)
//reset({ tipo: ty })
//alert(ty)
};

**Reply by bluebill1049** [reply]
have a look at this example: https://snack.expo.io/@bluebill1049/react-hook-form-v6
